app/src/main/java/com/github/swent/echo/compose/components/EventInfoSheet.kt

```diff
@@ -0,0 +1,195 @@
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Face
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.Text
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+/**
+ * Bottom sheet that shows the details of an event.
+ *
+ * @param showBottomSheet state to show the bottom sheet (mutable)
+ * @param eventTitle title of the event
+ * @param eventOrganization organization hosting the event
+ * @param eventDescription description of the event
+ * @param eventDateTime date and time of the event
+ * @param eventImage image id of the event
+ * @param eventPeople number of people who joined the event
+ * @param eventPeopleMax maximum number of people who can join the event
+ * @param onJoinButtonPressed callback to join the event
+ * @param onShowPeopleButtonPressed callback to show people who joined the event
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun EventInfoSheet(
+    showBottomSheet: MutableState<Boolean>,
+    eventTitle: String,
+    eventOrganization: String,
+    eventDescription: String,
+    eventDateTime: String,
+    eventImage: Int, // not sure how to handle images in the data system yet
+    eventPeople: Int,
+    eventPeopleMax: Int,
+    onJoinButtonPressed: () -> Unit,
+    onShowPeopleButtonPressed: () -> Unit
+) {
+    val sheetState = rememberModalBottomSheetState()
+    // set to false initially, will need to change it to true to show the bottom sheet (by
+    // clicking events)
+    if (showBottomSheet.value) {
+
+        ModalBottomSheet(
+            onDismissRequest = { showBottomSheet.value = false },
+            sheetState = sheetState,
+            containerColor = MaterialTheme.colorScheme.primaryContainer
+        ) {
+            // Sheet content
+            Box(
+                modifier =
+                    Modifier.padding(start = 20.dp, end = 20.dp, top = 0.dp, bottom = 32.dp)
+                        .width(360.dp)
+                        .height(375.dp)
```

Comment:
Use `fillMaxWidth()` and `fillMaxHeight(0.5f)` to fill the bottom half of the screen. This is the `PartiallyExpanded` state of the `ModalBottomSheet`. Apply those before the padding.

Reply:
that makes sense, I'll change that